var/spack/repos/builtin/packages/python/package.py

```diff
@@ -324,6 +332,105 @@ def filter_compilers(self, prefix):
+            exe = 'python2'
+        # If neither were found, try `python`
+        elif os.path.exists(os.path.join(self.prefix.bin, 'python')):
+            exe = 'python'
+        else:
+            msg = 'Cannot locate python executable in {0}'
+            raise RuntimeError(msg.format(self.prefix.bin))
+
+        # The python executable may be a symlink if it was installed
+        # with Homebrew. Since some packages try to determine the
+        # location of libraries and headers based on the path,
+        # return the realpath
+        return os.path.realpath(join_path(self.prefix.bin, exe))
+
+    @property
+    def home(self):
+        """Most of the time, ``PYTHONHOME`` is simply
+        ``spec['python'].prefix``. However, if the user is using an
+        externally installed python, it may be symlinked. For example,
+        Homebrew installs python in ``/usr/local/Cellar/python/2.7.12_2``
+        and symlinks it to ``/usr/local``. Users may not know the actual
+        installation directory and add ``/usr/local`` to their
+        ``packages.yaml`` unknowingly. Query the python executable to
+        determine exactly where it is installed."""
+
+        if self.spec.satisfies('@:2'):
+            print_statement = 'print sys.prefix'
+        else:
+            print_statement = 'print(sys.prefix)'
+
+        python = Executable(self.executable)
+        prefix = python('-c', 'import sys; {0}'.format(print_statement),
+                        output=str)
+
+        return Prefix(prefix)
+
+    @property
+    def libs(self):
+        libname = 'libpython{0}'.format(self.version.up_to(2))
+
+        # In Python 3, libraries have an 'm' at the end
+        if self.spec.satisfies('@3:'):
+            libname += 'm'
+
+        # Spack installs libraries into lib, except on openSUSE where it
+        # installs them into lib64. If the user is using an externally
+        # installed package, it may be in either lib or lib64, so we need
+        # to search both directories.
+        for directory in ['lib64', 'lib']:
+            root = join_path(self.home, directory)
+
+            # Prefer shared libraries
+            for shared in [True, False]:
+                libs = find_libraries(libname, root=root, shared=shared)
+
+                if libs:
+                    return libs
+
+        msg = 'Cannot locate python libraries in {0}/lib64 or {0}/lib'
+        raise RuntimeError(msg.format(self.home))
+
+    @property
+    def headers(self):
+        pythonver = 'python{0}'.format(self.version.up_to(2))
+
+        # In Python 3, the include directory has an 'm' at the end
+        if self.spec.satisfies('@3:'):
+            pythonver += 'm'
+
+        root = join_path(self.home, 'include', pythonver)
+        headers = find_headers('*.h', root=root)
+
+        if headers:
+            return headers
+
+        msg = 'Cannot locate python headers in {0}'
+        raise RuntimeError(msg.format(root))
```

Comment:
This error message will be helpful for users who use an externally installed system python but don't have the development headers installed.